packages/flutter_tools/templates/plugin/test/projectName_test.dart.tmpl

```diff
@@ -1,23 +1,27 @@
-import 'package:flutter/services.dart';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:{{projectName}}/{{projectName}}.dart';
+import 'package:{{projectName}}/{{projectName}}_platform_interface.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
 
-void main() {
-  const MethodChannel channel = MethodChannel('{{projectName}}');
+class Mock{{pluginDartClass}}Platform 
+    with MockPlatformInterfaceMixin
+    implements {{pluginDartClass}}Platform {
 
-  TestWidgetsFlutterBinding.ensureInitialized();
+  @override
+  Future<String?> getPlatformVersion() => Future.value('42');
+}
 
+void main() {
+  late {{pluginDartClass}} {{pluginClassLowerCamelCase}};
+  late Mock{{pluginDartClass}}Platform fakePlatform;
+  
   setUp(() {
-    channel.setMockMethodCallHandler((MethodCall methodCall) async {
-      return '42';
-    });
-  });
-
-  tearDown(() {
-    channel.setMockMethodCallHandler(null);
+    fakePlatform = Mock{{pluginDartClass}}Platform();
+    {{pluginDartClass}}Platform.instance = fakePlatform;
+    {{pluginClassLowerCamelCase}} = {{pluginDartClass}}();
   });
 
```

Comment:
We should add a test like this one:
https://github.com/flutter/plugins/blob/7686be7ecbe1cc049423715b5e2d897cb63a43c1/packages/file_selector/file_selector_platform_interface/test/file_selector_platform_interface_test.dart#L14-L16

as it's good to have that coverage to make sure the default behavior is as expected.